craft_cli/dispatcher.py

```diff
@@ -20,13 +20,16 @@
 import argparse
 import dataclasses
 import difflib
-from typing import Any, Callable, Literal, NamedTuple, NoReturn, Optional, Sequence
+from typing import TYPE_CHECKING, Any, Literal, NamedTuple, NoReturn, Optional
 
 from craft_cli import EmitterMode, emit
 from craft_cli.errors import ArgumentParsingError, ProvideHelpException
 from craft_cli.helptexts import HelpBuilder, OutputFormat
 from craft_cli.utils import humanize_list
 
+if TYPE_CHECKING:
+    from collections.abc import Callable
```

Comment:
I think we can keep this out of the `TYPE_CHECKING` guard. We're already importing `typing` anyway

Reply:
I think adding a noqa is fine, you can add a comment saying that the TYPE_CHECKING guard breaks sphinx doc generation
or just disable the ruff rule for craft-cli (probably better)